### ISSUE TITLE
ERROR: for relay Container "60f6f0106686" is unhealthy.

### ISSUE THREAD

**Issue by brunis** [getsentry/self-hosted#1235]
### Version

9.1.2

### Steps to Reproduce

1. ran ./install.sh
2. (upgraded to 10.0)
3. ran docker-compose up

### Expected Result

expected Sentry to run

### Actual Result

I receive this after all containers 
ERROR: for relay Container "60f6f0106686" is unhealthy.
ERROR: Encountered errors while bringing up the project.

How do i figure out what goes wrong in that particular container (it's sentry-web) ?

**Comment by brunis** [comment]
All sentry containers (except one) are in a constanct loop of restarting:
07d08460786a   getsentry/sentry:83b1380                 "/etc/sentry/entrypo…"   16 minutes ago   Restarting (0) 9 seconds ago                                   sentry_onpremise_ingest-consumer_1
6a54b5dc75e2   getsentry/sentry:83b1380                 "/etc/sentry/entrypo…"   16 minutes ago   Up 4 seconds                    9000/tcp                       sentry_onpremise_subscription-consumer-transactions_1
765c382b6543   getsentry/sentry:83b1380                 "/etc/sentry/entrypo…"   16 minutes ago   Restarting (0) 51 seconds ago                                  sentry_onpremise_cron_1
ec7329d42280   getsentry/sentry:83b1380                 "/etc/sentry/entrypo…"   16 minutes ago   Restarting (0) 58 seconds ago                                  sentry_onpremise_worker_1
2641d170404c   sentry-cleanup-self-hosted-local         "/entrypoint.sh '0 0…"   16 minutes ago   Up 16 minutes                   9000/tcp                       sentry_onpremise_sentry-cleanup_1
edf486633a2e   getsentry/sentry:83b1380                 "/etc/sentry/entrypo…"   16 minutes ago   Restarting (0) 51 seconds ago                                  sentry_onpremise_post-process-forwarder_1
60f6f0106686   getsentry/sentry:83b1380                 "/etc/sentry/entrypo…"   16 minutes ago   Restarting (0) 58 seconds ago                                  sentry_onpremise_web_1
1566024fc6de   getsentry/sentry:83b1380                 "/etc/sentry/entrypo…"   16 minutes ago   Restarting (0) 14 seconds ago                                  sentry_onpremise_subscription-consumer-events_1

**Comment by brunis** [comment]
these are the logs of all my sentry-* containers:

```
found /etc/sentry/docker-entrypoint.sh
Updating certificates in /etc/ssl/certs...
0 added, 0 removed; done.
Running hooks in /etc/ca-certificates/update.d...
done.
```

I must have done something wrong..

**Comment by aminvakil** [comment]
@brunis Try to install again in a clean state (using `reset.sh` or manually), then paste errors of `install.sh`.

**Comment by brunis** [comment]
> these are the logs of all my sentry-* containers:
> 
> ```
> found /etc/sentry/docker-entrypoint.sh
> Updating certificates in /etc/ssl/certs...
> 0 added, 0 removed; done.
> Running hooks in /etc/ca-certificates/update.d...
> done.
> ```

i had to manually copy docker-entrypoint.sh from a temp sentry container from a tip on SO:
docker run -d --name sentry-temp getsentry/sentry:10
docker cp sentry-temp:/etc/sentry/docker-entrypoint-sh sentry

**Comment by brunis** [comment]
> @brunis Try to install again in a clean state (using `reset.sh` or manually), then paste errors of `install.sh`.

I have an existing (manually upgraded) postgres9.6 db.. the reset.sh script says it's high destructive and i can see some volume removal..

# Remove any remaining (likely external) volumes with name matching 'sentry-.*'.
^^
That doesn't sound like my data will survive?

**Comment by brunis** [comment]
Zero change:
ERROR: for relay  Container "6ecacaf57991" is unhealthy.
ERROR: Encountered errors while bringing up the project.

I'm guessing it's the docker-entrypoint.sh i had to manually copy into ./sentry/ that is causing the issue..

**Comment by aminvakil** [comment]
```
 docker run -d --name sentry-temp getsentry/sentry:10
 docker cp sentry-temp:/etc/sentry/docker-entrypoint-sh sentry
```
Why are you running sentry like this?
Please follow [official documentation](https://develop.sentry.dev/self-hosted/#getting-started).

**Comment by brunis** [comment]
> ```
>  docker run -d --name sentry-temp getsentry/sentry:10
>  docker cp sentry-temp:/etc/sentry/docker-entrypoint-sh sentry
> ```
> 
> Why are you running sentry like this? Please follow [official documentation](https://develop.sentry.dev/self-hosted/#getting-started).

I'm not running it like that, it was an SO article suggesting to get them that way, cause they are missing. This is what i get without them:
`Error response from daemon: failed to create shim: OCI runtime create failed: container_linux.go:380: starting container process caused: exec: "/etc/sentry/entrypoint.sh": stat /etc/sentry/entrypoint.sh: no such file or directory: unknown`

**Comment by aminvakil** [comment]
I don't know how copying entrypoint to docker after it's already ran gonna help, anyway I'd suggest cloning repository again, and try installing in a clean state, or you should offer some more info. AFAICS `sentry/entrypoint.sh` is present in repository.

**Comment by brunis** [comment]
> I don't know how copying entrypoint to docker after it's already ran gonna help, anyway I'd suggest cloning repository again, and try installing in a clean state, or you should offer some more info. AFAICS `sentry/entrypoint.sh` is present in repository.

yes, you are right, i accidentally deleted that now when i tried the reset.sh, as you can see from the code i was copying docker-entrypoint.sh ..not entrypoint.sh
entrypoint.sh is trying to 'source /docker-entrypoint.sh' ..that is the one missing..

./install.sh now gives me:
/etc/sentry/entrypoint.sh: line 20: /docker-entrypoint.sh: No such file or directory

**Comment by brunis** [comment]
> I don't know how copying entrypoint to docker after it's already ran gonna help, anyway I'd suggest cloning repository again, and try installing in a clean state, or you should offer some more info. AFAICS `sentry/entrypoint.sh` is present in repository.

i'm doing the 9.1.2 -> v10 first..  entrypoint.sh is also missing from TAG 10.0 ..

**Comment by aminvakil** [comment]
> > I don't know how copying entrypoint to docker after it's already ran gonna help, anyway I'd suggest cloning repository again, and try installing in a clean state, or you should offer some more info. AFAICS `sentry/entrypoint.sh` is present in repository.
> 
> i'm doing the 9.1.2 -> v10 first.. entrypoint.sh is also missing from TAG 10.0 ..

Because it does not need it https://github.com/getsentry/self-hosted/blob/10.0.0/sentry/Dockerfile

**Comment by brunis** [comment]
> I don't know how copying entrypoint to docker after it's already ran gonna help, anyway I'd suggest cloning repository again, and try installing in a clean state, or you should offer some more info. AFAICS `sentry/entrypoint.sh` is present in repository.

I just tried a full clean ./install.sh of 10.0 ..which works, as a clean install, but mapping my old sentry-postgres and ./install.sh again makes it fail, it tries to connect to hostname "postgres", which fails.

I then tried a full clean ./install.sh of 21.5 (which is the max recommended 9.1.2 upgrade) it also works as a clean install. And installing ./install.sh again with a sentry-postgres volume map to my old v9.1.2 sentry (pg 9.5) data actually upgrades it this time.. the whole migration takes a long time (it's about 1.2gb of data). Then the installer offers to create an account .. and when i log in it's a cleaned database with the default sentry project.

So, both clean install and upgrade runs and "works" .. but migrated data is ignored or has disappeared, so what to do now? ..
